### DISCUSSION TITLE
rust wasm file schema version: 0.2.88 this binary schema version: 0.2.87

### DISCUSSION THREAD

**Discussion by lhjok** [wasm-bindgen/wasm-bindgen#3684] [Q&A]
### Steps to Reproduce

If applicable, add a link to a test case (as a zip file or link to a repository we can clone).

### Expected Behavior

During compilation

### Actual Behavior

2023-11-04T04:50:55.757707Z  INFO calling wasm-bindgen for Leptos-rs
error: 

it looks like the Rust project used to create this wasm file was linked against
version of wasm-bindgen that uses a different bindgen format than this binary:

  rust wasm file schema version: 0.2.88
     this binary schema version: 0.2.87 (f0a8ae3b9)

Currently the bindgen format is unstable enough that these two schema versions
must exactly match. You can accomplish this by either updating this binary or 
the wasm-bindgen dependency in the Rust project.

You should be able to update the wasm-bindgen dependency with:

    cargo update -p wasm-bindgen --precise 0.2.87 (f0a8ae3b9)

don't forget to recompile your wasm file! Alternatively, you can update the 
binary with:

    cargo install -f wasm-bindgen-cli --version 0.2.88

if this warning fails to go away though and you're not sure what to do feel free
to open an issue at https://github.com/rustwasm/wasm-bindgen/issues!

2023-11-04T04:50:55.928656Z ERROR ❌ error
error from HTML pipeline

Caused by:
    0: error from asset pipeline
    1: wasm-bindgen call returned a bad status
Error: error from HTML pipeline

Caused by:
    0: error from asset pipeline
    1: wasm-bindgen call returned a bad status

**Comment by daxpedda** [comment]
Did you already successfully update `wasm-bindgen-cli`?
If you `wasm-bindgen -V` what version does it return?
If you are using `cargo leptos`, you have to update that as well: `cargo install -f cargo-leptos`.

**Reply by lhjok** [reply]
cargo-leptos has not been updated

**Reply by lhjok** [reply]
I build using Trunk

**Reply by daxpedda** [reply]
> cargo-leptos has not been updated

Yes, but if you rebuild it, it uses the newest dependencies, which would resolve the problem.

> I build using Trunk

AFAIK `trunk` uses your locally installed `wasm-bindgen-cli`.
If you `wasm-bindgen -V` what version does it return?

**Reply by lhjok** [reply]
versions match

**Reply by AlexErrant** [reply]
It may be due to the lockfile. Either delete and recreate, or try pinning to an exact version: `"=0.2.87"`. Emphasis on the `=`. I ran into a similar issue [here](https://github.com/tweag/rust-wasm-threads/issues/5).

**Reply by daxpedda** [reply]
> versions match

Match which one? It would be helpful if you provide as much information as possible, e.g. please share the output of `wasm-bindgen -V`.

**Comment by NicTanghe** [comment]
i have a provblem where the bindgen is 0.2.93 and the binairy says it is 0.2.96 but it is 
14f7c3d3bbb4:/homebrew/TFX# wasm-bindgen -V
wasm-bindgen 0.2.93

I also reinstalled cargo leptos and got same results.

**Reply by daxpedda** [reply]
How did you determine that you are depending on `wasm-bindgen` v0.2.93?
Cargo by default uses [caret requirement](https://doc.rust-lang.org/1.83.0/cargo/reference/specifying-dependencies.html#caret-requirements), so if you use `wasm-bindgen = "0.2.93"`, it will actually pull in the latest version.

The best way to check is to look into your `Cargo.lock` file.

**Comment by Davud345** [comment]
Compilation fails due to a mismatch between the wasm-bindgen binary version and the Rust project’s wasm-bindgen dependency.

Steps to Fix:

Update the Rust project's dependency:

cargo update -p wasm-bindgen --precise 0.2.87

Or update the wasm-bindgen-cli binary:

cargo install -f wasm-bindgen-cli --version 0.2.88

Recompile the project and ensure the versions match.
For further help, refer to wasm-bindgen GitHub issues 

Bonus Tip: Fruit Op Het Werk 🍎
Boost your debugging energy with a fresh fruit snack at work!
